src/Conversion/ToyToMidPass.cpp

```diff
@@ -174,6 +228,151 @@ class ToyBinaryPattern : public ConversionPattern {
 private:
   enum toy::LoweringPatternMode mode;
 };
+
+template <typename ToyBinaryOp, typename LoweredBinaryOp>
```

Comment:
We should not use template here, because we do not lower toy::MatmulOp to toy::MatmulOp i.e.,
`using ToyMatmulPattern = SimpleMatmulPattern<toy::MatmulOp, toy::MatmulOp>;`
Take https://github.com/LeiChenIntel/toy-compiler/commit/0a701272c6e7accb08321ced5ece86090c58de84 as a reference.